weave/flow/eval2.py

```diff
@@ -0,0 +1,200 @@
+
+            # The following section is a "hacky" way to create Model and Evaluation
+            # objects that "look right" to our object saving system.
+
+            # --- Setup the model object ---
+            @weave.op
+            def predict(self: Model, inputs: dict) -> Any:
+                # Get the output from the context variable
+                return current_output.get()
+
+            self._pseudo_model.__dict__["predict"] = MethodType(
+                predict, self._pseudo_model
+            )
+
+            # --- Setup the evaluation object ---
+            @weave.op(name="Evaluation.evaluate")
+            def evaluate(self: Evaluation, model: Model) -> None: ...
+
+            @weave.op
+            def predict_and_score(self: Evaluation, model: Model, inputs: dict) -> dict:
+                model_output = model.get_infer_method()(inputs)
+                return {
+                    "model_output": model_output,
+                    "scores": NOT_DEFINED,
+                    "model_latency": NOT_DEFINED,
+                }
+
+            @weave.op
+            def summarize(self: Evaluation) -> dict:
+                return cast(dict, current_summary.get())
+
+            self._pseudo_evaluation.__dict__.update(
+                {
+                    "evaluate": MethodType(evaluate, self._pseudo_evaluation),
+                    "predict_and_score": MethodType(
+                        predict_and_score, self._pseudo_evaluation
+                    ),
+                    "summarize": MethodType(summarize, self._pseudo_evaluation),
+                }
+            )
+
+            self._starting_stack = call_context.get_call_stack()
+
+            # Create the evaluation call
+            wc = require_weave_client()
+            self._evaluate_call = wc.create_call(
+                display_name=default_evaluation_display_name,
+                op=self._pseudo_evaluation.evaluate,
+                inputs={
+                    "self": self._pseudo_evaluation,
+                    "model": self._pseudo_model,
+                },
+            )
+            assert self._evaluate_call is not None
+            call_context.push_call(self._evaluate_call)
+
+        # Make the prediction call
+        with set_current_output(output):
+            _, predict_and_score_call = self._pseudo_evaluation.predict_and_score.call(
+                self._pseudo_evaluation, self._pseudo_model, inputs
+            )
+
+        # Get the model output from the call result
+        model_output = predict_and_score_call.output.get("model_output")
+
+        assert self._evaluate_call is not None
+        return BetaPredictionLogger(
+            model_id="123",
+            inputs=inputs,
+            output=model_output,
+            hack_reference_to_predict_and_score=predict_and_score_call,
+            hack_reference_to_evaluate=self._evaluate_call,
+        )
+
+    def log_summary(self, summary: dict) -> None:
```

Comment:
it would be nice if the summary could support a function that accepted prior rows

Reply:
I think we can leave this up to the user for now.  The user has full control and can implement their own roll-up func for summary